internal/test/framework/framework.go

```diff
@@ -1,81 +1,86 @@
 	// OSConfigClient is the OpenShift config client, we will use to query the OpenShift api object status
 	OSConfigClient *configclient.Clientset
 }
 
-// Setup sets up the Windows node so that it can join the existing OpenShift cluster
-// TODO: move this to return error and do assertions there
-func (f *TestFramework) Setup() {
-	if err := f.createWindowsVM(); err != nil {
-		log.Fatalf("failed to create Windows VM: %v", err)
+// initCIvars gathers the values of the environment variables which configure the test suite
+func initCIvars() error {
+	kubeconfig = os.Getenv("KUBECONFIG")
+	if kubeconfig == "" {
+		return fmt.Errorf("KUBECONFIG environment variable not set")
 	}
-	// TODO: Add some options to skip certain parts of the test
-	if err := f.setupWinRMClient(); err != nil {
-		log.Fatalf("failed to setup winRM client for the Windows VM: %v", err)
+	awsCredentials = os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+	if awsCredentials == "" {
+		return fmt.Errorf("AWS_SHARED_CREDENTIALS_FILE environment variable not set")
 	}
-	// Wait for some time before starting configuring of ssh server. This is to let sshd service be available
-	// in the list of services
-	// TODO: Parse the output of the `Get-Service sshd, ssh-agent` on the Windows node to check if the windows nodes
-	// has those services present
-	time.Sleep(time.Minute)
-	if err := f.configureOpenSSHServer(); err != nil {
-		log.Fatalf("failed to configure OpenSSHServer on the Windows VM: %v", err)
+	artifactDir = os.Getenv("ARTIFACT_DIR")
+	if awsCredentials == "" {
+		return fmt.Errorf("ARTIFACT_DIR environment variable not set")
 	}
-	if err := f.createRemoteDir(); err != nil {
-		log.Fatalf("failed to create remote dir with error: %v", err)
+	privateKeyPath = os.Getenv("KUBE_SSH_KEY_PATH")
+	if privateKeyPath == "" {
+		return fmt.Errorf("KUBE_SSH_KEY_PATH environment variable not set")
 	}
-	if err := f.getSSHClient(); err != nil {
-		log.Fatalf("failed to get ssh client for the Windows VM created: %v", err)
+	return nil
+}
+
+// Setup creates and initializes a variable amount of Windows VMs
+func (f *TestFramework) Setup(vmCount int) error {
+	// Use Windows 2019 server image with containers in us-east1 zone for CI testing.
+	// TODO: Move to environment variable that can be fetched from the cloud provider
+	// The CI-operator uses AWS region `us-east-1` which has the corresponding image ID: ami-0b8d82dea356226d3 for
+	// Microsoft Windows Server 2019 Base with Containers.
+	imageID := "ami-0b8d82dea356226d3"
+	// Using an AMD instance type, as the Windows hybrid overlay currently does not work on on machines using
+	// the Intel 82599 network driver
+	instanceType := "m5a.large"
+	if err := initCIvars(); err != nil {
+		return fmt.Errorf("unable to initialize CI variables: %v", err)
+	}
+	f.WinVMs = make([]WindowsVM, vmCount)
+	// TODO: make them run in parallel: https://issues.redhat.com/browse/WINC-178
+	for i := 0; i < vmCount; i++ {
+		var err error
```

Comment:
Do you want to return immediately or consolidate errors for all the Windows nodes and return?

Reply:
I want to return immediately. When we make the setup run in parallel the errors will get automatically have to be consolidated.